site/_posts/2020-07-22-securely-implement-oauth-vuejs.md

```diff
@@ -0,0 +1,1213 @@
+
+## Setting user data from Vue
+
+Now that we have created our route for updating user data, let’s create a `text-area` in our Vue app. We will type the user data in that, which will be sent to the server when the `Submit` button is clicked.
+
+In `client/src` create a new file named `Update.vue` and add the following to it:
+
+```html
+<template>
+  <form>
+    <textarea
+      v-model="userData"
+      placeholder="Update FusionAuth user data."
+    ></textarea>
+    <button type="submit" class="button">Submit</button>
+  </form>
+</template>
+<script>
+export default {
+  name: "Update",
+  data() {
+    return {
+      userData: "",
+    };
+  },
+</script>
+<style>
+textarea {
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+}
+button {
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 5px;
+}
+</style>
+```
+
+One of the cool features of Vue is that by using `v-model="userData"` and initializing `userData=""` in the `data()` function, it creates a two-way data binding between `textarea` and `userData`.
+
+We can now access whatever we type in `textarea` in `userData`. You can [read more about it here](https://vuejs.org/v2/guide/forms.html).
+
+Then add this component to `App.vue` . It does not make sense to show this component when the user is not logged in. To hide it, just add `v-if="email"` with this component. Like some of our other components it will check to see if `email` is present or not, i.e., if user is logged in or not.
+
+```html
+<Update v-if="email" />
+```
+
+We still haven’t configured the `Submit` button. Let's do so to send whatever we type in `textarea` to our server to be stored. For this, we will create a function `update` inside the `methods()` section.
+
+```javascript
+//...
+methods: {
+  update: function() {
+    fetch(`http://localhost:9000/set-user-data`, {
+      credentials: "include",
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        userData: this.userData,
+      }),
+    }).catch((err) => {
+        console.log(err);
+      });
+      this.userData=''
+  },
+},
+//...
+```
+
+In the above function, we use `fetch()` to **POST** JSON-encoded data. If you are familiar with `fetch()` API, you will see that this is a simple **POST** request using `fetch()`, nothing fancy. You can [read more about it here](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).
+
+Once we have sent `userData` to our server, we empty the `textarea` using `this.userData=''`, since it is a two-way binding.
+
+And now to bind this function to `submit` event we will add the following to the `form` tag.
+
+```html
```

Comment:
This code type should use whatever language supports the Vue syntax. Otherwise, it renders as an error with the "@" attribute.

Reply:
Yup, it's `vue`. Here's the list of all the formatters supported by `rouge`, the jekyll code highlighter: https://github.com/rouge-ruby/rouge/tree/master/lib/rouge/lexers